packages/core/src/tools/promise.js

```diff
@@ -23,7 +23,7 @@ const enrichErrorMessage = (err, frameStack) => {
 
 // Because reject callbacks stack, we have to skip errors we've already seen.
 const contextifyOnce = (err, attachErrorTrace, frameStack) => {
-  if (!err || err.isContextified) {
+  if (!err || err.doNotContextify || err.isContextified) {
```

Comment:
This makes sure that we don't append `What happened:` to the `JSON.stringify`-ed message of the new errors.